### ISSUE TITLE
resource: support alternate writer format for `find`

### ISSUE THREAD

**Issue by SteVwonder** [flux-framework/flux-sched#736]
Currently, `find` only supports RV1_nosched as the writer format. It would be nice to be able to customize the writer format (potentially even per-request).

**Comment by dongahn** [comment]
Yes, this can be done fairly easily by allowing the predefined writer name as an optional argument.

The first real use case for `find` was for `flux resource list` which requires `rv1_nosched` but the code is there to emit this in other formats.

My guess is your use case is to emit this in `JGF` for your multi storage work?

**Comment by SteVwonder** [comment]
> My guess is your use case is to emit this in JGF for your multi storage work?

Yeah, it's been really useful for debugging unsatisfiable jobspecs.  I have a patch in my current commit stack that sets the writer for `find` to the same as the one for `match`, but I think it could easily be extended to be an argument in the `find` RPC.  So once the L2 is done, I can take a shot at that.
